mautic/mautic
# UI warnings when unpublish associated segment

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | https://github.com/mautic/mautic-documentation/pull/207
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Currently if a segment filter is using another segment as a source, and that source is unpublished, the segment will disappear from the segment membership filter.

This is expected behavior as unpublished segments are not available to segment filters, however this can cause issue when the filter ends up empty, leading to all contacts being added to the segment, which can be resource intensive.

**Solution :** 
Now we will warn the user that the segment is used as a filter within another segment and block them from unpublishing


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1. Create a segment - S1
2. Create another segment - S2
3. In S2 segment, add segment membership filter and use segment S1 as value
4. Save & Close
5. Now Unpublish segment S1
6. S1 should not be unplublish and should show a warning message.

#### Other areas of Mautic that may be affected by the change:
1.
2.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
